File: 53._Maximum_Subarray/code.cpp
class Solution {
public:
    int maxSubArray(vector<int>& nums) {
        int current = 0;
        int maxNum = nums[0];
        
        for(int i = 0; i < nums.size(); i++) {
            current = max(nums[i] + current, nums[i]);
            maxNum = max(maxNum, current);
        }
        return maxNum;
    }
};

File: 53._Maximum_Subarray/Swift/code.swift
class Solution {
    func maxSubArray(_ nums: [Int]) -> Int {
        var maxNum: Int = nums[0]
        var current: Int = 0
        
        for i in nums {
            current = max(i + current, i)
            maxNum = max(maxNum, current)
        }
        return maxNum
    }
}
